common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -327,6 +325,56 @@ def _student_view(self, context, banner_text=None):
 
         return fragment
 
+    def _is_gate_fulfilled(self):
+        """
+        Determines if this section is a prereq and has any unfulfilled milestones.
+
+        Returns:
+            True if section has no unfufilled milestones or is not a prerequiste.
+            False otherwise
+        """
+        gating_service = self.runtime.service(self, 'gating')
+        if gating_service:
+            fulfilled = gating_service.is_gate_fulfilled(
+                self.course_id, self.location, self.runtime.user_id
+            )
+            return fulfilled
+
+        return True
+
+    def _is_prereq_required(self):
+        """
+        Checks whether a prerequiste is required for this Section
+
+        Returns:
+            milestone if a prereq is required, False otherwise
+        """
+        gating_service = self.runtime.service(self, 'gating')
+        if gating_service:
+            milestone = gating_service.is_prereq_required(
+                self.course_id, self.location, 'requires'
+            )
+            return milestone
```

Comment:
Nit: Based on the name, I'd expect this method to return a boolean. Having it return a boolean in one branch and a milestone object in another is unusual. A more common pattern would be to have a name like `required_prereq()` that returns the object or `None` if there is no required prerequisite.

Reply:
@ormsbee 
changed to use required_prereq() as that makes more sense as you pointed out